api.planx.uk/server.ts

```diff
@@ -309,12 +310,31 @@ app.post("/pay/:localAuthority", (req, res, next) => {
   const isSupported =
     process.env[`GOV_UK_PAY_TOKEN_${req.params.localAuthority.toUpperCase()}`];
 
+  const flowId = req.query?.flowId as string | undefined;
+  const sessionId = req.query?.sessionId as string | undefined;
+  const teamSlug = req.params.localAuthority;
+
   if (isSupported) {
     // drop req.params.localAuthority from the path when redirecting
     // so redirects to plain [GOV_UK_PAY_URL] with correct bearer token
     usePayProxy(
       {
         pathRewrite: (path) => path.replace(/^\/pay.*$/, ""),
+        selfHandleResponse: true,
+        onProxyRes: responseInterceptor(
+          async (responseBuffer, _proxyRes, _req, _res) => {
+            const responseString = responseBuffer.toString("utf8");
+            const govUkResponse = JSON.parse(responseString);
+            await logPaymentStatus({
+              sessionId,
+              flowId,
+              teamSlug,
+              govUkResponse,
+            });
+            return JSON.stringify(govUkResponse);
+            return responseBuffer;
+          }
+        ),
```

Comment:
thanks for making this adjustment to the API - this implementation feels a lot cleaner to me, and now very aligned to how we update our other audit tables, than adding a new method to the Preview store

apologies again if it was frustrating feedback to get in review, rather than as early context in the ticket itself - but i really think this is worth it, especially now being able to have this mutation on _both_ pay API endpoints which is of huge value